src/lib/Microsoft.Fx.Portability.Offline/OfflineApiPortService.cs

```diff
@@ -35,98 +37,67 @@ public OfflineApiPortService(IApiCatalogLookup lookup, IRequestAnalyzer requestA
             _apiRecommendations = apiRecommendations;
         }
 
-        public Task<ServiceResponse<IEnumerable<AvailableTarget>>> GetTargetsAsync()
+        public Task<IEnumerable<AvailableTarget>> GetTargetsAsync()
         {
             var targets = _lookup
                 .GetPublicTargets()
                 .Select(target => new AvailableTarget { Name = target.Identifier, Version = target.Version, IsSet = _defaultTargets.Contains(target) })
-                .OrderBy(t => t.Name, StringComparer.OrdinalIgnoreCase);
-
-            var response = new ServiceResponse<IEnumerable<AvailableTarget>>(targets);
+                .OrderBy(t => t.Name, StringComparer.OrdinalIgnoreCase)
+                .AsEnumerable();
 
-            return Task.FromResult(response);
+            return Task.FromResult(targets);
         }
 
-        public Task<ServiceResponse<AnalyzeResponse>> SendAnalysisAsync(AnalyzeRequest a)
+        public Task<IEnumerable<ResultFormatInformation>> GetResultFormatsAsync()
         {
-            var response = _requestAnalyzer.AnalyzeRequest(a, Guid.NewGuid().ToString());
-            var serviceResponse = new ServiceResponse<AnalyzeResponse>(response);
+            var formats = _reportWriters.Select(r => r.Format);
 
-            return Task.FromResult(serviceResponse);
+            return Task.FromResult(formats);
         }
 
-        public async Task<ServiceResponse<IEnumerable<ReportingResultWithFormat>>> SendAnalysisAsync(AnalyzeRequest a, IEnumerable<string> formats)
+        public Task<ResultFormatInformation> GetDefaultResultFormatAsync()
         {
-            var response = _requestAnalyzer.AnalyzeRequest(a, Guid.NewGuid().ToString());
-
-            if(!formats?.Any() ?? true)
-            {
-                var defaultFormat = await GetDefaultResultFormatAsync();
-                formats = new[] { defaultFormat.Response.DisplayName };
-            }
-
-            var formatSet = new HashSet<string>(formats, StringComparer.OrdinalIgnoreCase);
-
-            var result = new List<ReportingResultWithFormat>();
-
-            foreach (var writer in _reportWriters.Where(w => formatSet.Contains(w.Format.DisplayName)))
-            {
-                using (var ms = new MemoryStream())
-                {
-                    writer.WriteStream(ms, response);
-
-                    result.Add(new ReportingResultWithFormat
-                    {
-                        Format = writer.Format.DisplayName,
-                        Data = ms.ToArray()
-                    });
-                }
-            }
+            var format = new JsonReportWriter().Format;
```

Comment:
you can have this as a private static readonly JsonReportWriter so it is only created once and reused.

Reply:
True, but this function is only invoked once.